File: tutorials/testing/java/osm_snippet.java
import bridges.connect.Bridges;
import bridges.connect.DataSource;
import bridges.data_src_dependent.OsmData;
import bridges.data_src_dependent.OsmVertex;
import bridges.data_src_dependent.OsmEdge;


// An example program fragment to illustrate how to retrieve Open Street map  data.
// The program rertrieves the map of UNC Charlotte campus, prints the number of vertices
// and edges, and the location of the first vertex

public class osm_snippet {
	public static void main(String[] args) throws Exception {

		//create the Bridges object
		Bridges bridges = new Bridges(YOUR_ASSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY");

		DataSource  ds = bridges.getDataSource();
		//		OsmData osm_data = ds.getOsmData("Charlotte, North Carolina", "default");
		OsmData osm_data = ds.getOsmData(41.03133177632377, -98.02593749997456,
				42.008577297430456, -97.94531249997696);



		OsmVertex[] vertices = osm_data.getVertices();
		OsmEdge[] edges = osm_data.getEdges();

		System.out.println("Number of Vertices [Charlotte]:" + vertices.length);
		System.out.println("Number of Edges [Charlotte]:" + edges.length);

		// get cartesian coordinate  location of first vertex
		double[] coords = osm_data.getVertices()[0].getCartesian_coord();
		System.out.println ("Location of first vertex [Cartesian Coord]: " +  coords[0] + ","
			+ coords[1]);
	}
}

File: tutorials/testing/c++/osm_snippet.cpp
#include "Bridges.h"
#include "DataSource.h"
#include "GraphAdjList.h"
#include <iostream>
#include <string>
#include <unordered_map>
#include <fstream>
#include "data_src/OSMData.h"
#include "data_src/OSMVertex.h"
#include "data_src/OSMEdge.h"

using namespace std;
using namespace bridges;

int main(int argc, char **argv) {

	// create Bridges object
	Bridges bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY");

	DataSource ds(bridges);
//	OSMData osm_data = ds.getOSMData("Charlotte, North Carolina");
	OSMData osm_data = ds.getOSMData(41.03133177632377, -98.02593749997456, 
						42.008577297430456, -97.94531249997696);
	
	vector<OSMVertex> vertices = osm_data.getVertices();
	vector<OSMEdge> edges = osm_data.getEdges();

	double coords[2];
	cout << "Number of Vertices [Charlotte]:" << vertices.size() << endl;
	cout << "Number of Edges [Charlotte]:" << edges.size() << endl;

	// get cartesian coordinate  location of first vertex
	osm_data.getVertices()[0].getCartesianCoords(coords);
	cout << "Location of first vertex [Cartesian Coord]: " <<  coords[0] << ","
		<< coords[1] << endl;
	return 0;
}
